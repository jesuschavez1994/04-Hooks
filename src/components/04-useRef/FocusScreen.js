import React, { useRef } from 'react'

export const FocusScreen = () => {

    const inputRef = useRef()

    const handleClick = () => {
        inputRef.current.select();
    }

    return (
        <>
            <h1>FocusScreen</h1>
            <hr />

            <input 
                ref={ inputRef }
                className="form-control"
                placeholder="Name"
            />

            <button className="btn btn-primary mt-5" onClick={ handleClick }>
                Focus
            </button>
        </>
    )
}
