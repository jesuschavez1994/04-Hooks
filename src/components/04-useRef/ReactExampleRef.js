import React, { useState } from 'react'
import { MultipleCusttomHook } from '../03-examples/MultipleCusttomHook'


export const ReactExampleRef = () => {

    const [show, setShow] = useState(false);

    return (
        <>
            <h1>ReactExampleRef</h1>
            <hr />

            { show && <MultipleCusttomHook /> }

            <button className="btn btn-primary mt-5" onClick={ () => {
                setShow(!show);
            }}>
                Show
            </button>
            
        </>
    )
}
