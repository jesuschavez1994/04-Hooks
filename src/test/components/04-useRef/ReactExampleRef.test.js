import { shallow } from "enzyme";
import { ReactExampleRef } from "../../../components/04-useRef/ReactExampleRef";
import '@testing-library/jest-dom';

describe('Pruebas en <ReactExampleRef />', () => {
    
    const wrapper = shallow(<ReactExampleRef />);

    test('debe de mostrar correctamente ', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCusttomHook').exists()).toBe(false);
    });

    test('debe de mostrat <ReactExampleRef /> ', () => {
        wrapper.find('button').simulate('click');
        expect(wrapper.find('MultipleCusttomHook').exists()).toBe(true);
    });
    

})
