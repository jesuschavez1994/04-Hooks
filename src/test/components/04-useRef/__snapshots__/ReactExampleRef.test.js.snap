// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ReactExampleRef /> debe de mostrar correctamente  1`] = `
<Fragment>
  <h1>
    ReactExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show
  </button>
</Fragment>
`;

exports[`Pruebas en <ReactExampleRef /> should  1`] = `
<Fragment>
  <h1>
    ReactExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show
  </button>
</Fragment>
`;
